server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -391,8 +391,10 @@ public String executor() {
                 new InitializeExtensionRequest(transportService.getLocalNode(), extension),
                 initializeExtensionResponseHandler
             );
-            // TODO: make asynchronous
-            inProgressFuture.get(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+            inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+            if (inProgressFuture.isCompletedExceptionally()) {
```

Comment:
If I understand this correctly, we are calling `get()` here to force an exception to be thrown based on the completion (e.g., timeout or interrupted or cancelled).   I don't understand the exception catch block below this, however; we are catching and rethrowing the same exception. Why not just log the error here?

Reply:
That catches any extensions that may happen before .orTimeout is called